### ISSUE TITLE
imgui-sdl2 + specs

### ISSUE THREAD

**Issue by loops-and-spells** [michaelfairley/rust-imgui-sdl2#5]
It seems that passing the event_pump isn't very thread safe, I managed to get everything working great by mapping the sdl2 events to a HashMap structure and pushing it as resource state and modifying the handle_event and frame signatures.

It would be nice be a bit more data driven for the input states over passing a ref to the pump.
Possibly another function that accepts a struct, and another that accepts a pump.

I may be wrong, but I couldn't find a solution otherwise.

**Comment by michaelfairley** [comment]
Oh! Interesting. I had not planned on someone wanting to do their GUI work in another thread, but that makes total sense.

I pushed c8d0851f6279646ac43129467ea8afaceab3d157 and released 0.4.0 which has `frame` take `MouseState` (which is `Send`) instead of `EventPump`, and I opened up https://github.com/Rust-SDL2/rust-sdl2/pull/838 to see if `Event` can get made `Send` so `handle_event` would work as well.
